### PR TITLE
dev/core#560 Replace instances of CRM_Core_Error::fatal with Exceptio…

### PR DESCRIPTION
**Pull request by seamuslee001**
…ns or Status bounces in SMS processing and appropriately handle error after

Overview
----------------------------------------
This replaces instances of `CRM_Core_Error::fatal` in the SMS subsystem with CRM_Core_Exceptions or CRM_Core_Error::statusBounce as appropriate and updates the relevant functions to handle the change in error handling

Before
----------------------------------------
Legacy function used

After
----------------------------------------
Modern functions used

ping @eileenmcnaughton @mattwire

### CODE REVIEW

File: CRM/SMS/Form/Provider.php [hunk 1]
```diff
@@ -113,8 +113,12 @@ public function setDefaultValues() {
     $name = CRM_Utils_Request::retrieve('key', 'String', $this, FALSE, NULL);
     if ($name) {
       $defaults['name'] = $name;
-      $provider = CRM_SMS_Provider::singleton(['provider' => $name]);
-      $defaults['api_url'] = $provider->_apiURL;
+      try {
```
Comment (eileenmcnaughton) [inline CRM/SMS/Form/Provider.php]: So my hesitation is around this one & the next one which adds handling. The previous behaviour here was a fatal when the provider was invalid. Without the catch here the form layer would give the equivalent of that - but with the catch it seems an obscure edge case bug would be ignored

File: CRM/SMS/Page/Callback.php [hunk 2]
```diff
@@ -17,7 +17,13 @@
 class CRM_SMS_Page_Callback {
 
   public function run() {
-    $provider = CRM_SMS_Provider::singleton($_REQUEST);
+    try {
+      $provider = CRM_SMS_Provider::singleton($_REQUEST);
+    }
+    catch (CRM_Core_exception $e) {
```
Comment (eileenmcnaughton) [inline CRM/SMS/Page/Callback.php]: Also here - we are adding handling for what appears to be an obscure edge case bug where the no-change would be to leave it uncaught

**Comment by seamuslee001** [comment]
ok @eileenmcnaughton i have reverted the changes in those 2 places

**Comment by eileenmcnaughton** [comment]
looks good now

**Pull request merged**